### PR TITLE
Fix excluding pokemon with form

### PR DESCRIPTION
**Pull request by Mygod**
untested change ahead

### CODE REVIEW

File: src/views/index-js.mustache [hunk 1]
```diff
@@ -2633,13 +2633,17 @@ function getPokemonPopupContent (pokemon) {
             }
         });
     }
+    let pokemonIdString = '' + pokemon.pokemon_id + '-' + pokemon.form_id;
```
Comment (versx) [inline src/views/index-js.mustache:2636]:
```suggestion
    let pokemonIdString = '' + pokemon.pokemon_id + '-' + pokemon.form;
```
Even after working on RDM for almost 2 years I always thought it should be `form_id`, but.... it's just `form` for pokemon lol

File: src/views/index-js.mustache [hunk 2]
```diff
@@ -2633,13 +2633,17 @@ function getPokemonPopupContent (pokemon) {
...
+        pokemonIdString = pokemon.pokemon_id;
+    }
     content +=
         '</div>' +
     '</div>' + // END 3RD ROW
     '<br>' +
     '<div class="text-center">' +
         '<a id="h' + pokemon.id + '" title="Hide Pokemon" href="#" onclick="setIndividualPokemonHidden(\'' + pokemon.id + '\');return false;"><b>[Hide]</b></a>&nbsp;' +
-        '<a title="Filter Pokemon" href="#" onclick="addPokemonFilter(' + pokemon.pokemon_id + ', false);return false;"><b>[Exclude]</b></a>' +
+        '<a title="Filter Pokemon" href="#" onclick="addPokemonFilter(\'' + pokemonIdString + '\', ' + pokemon.pokemon_id + ', ' + pokemon.form_id + ', false);return false;"><b>[Exclude]</b></a>' +
```
Comment (versx) [inline src/views/index-js.mustache:2646]:
```suggestion
        '<a title="Filter Pokemon" href="#" onclick="addPokemonFilter(\'' + pokemonIdString + '\', ' + pokemon.pokemon_id + ', ' + pokemon.form + ', false);return false;"><b>[Exclude]</b></a>' +
```

File: src/views/index-js.mustache [hunk 3]
```diff
@@ -2691,12 +2695,12 @@ function setIndividualPokemonHidden (id) {
...
 // eslint-disable-next-line no-unused-vars
-function addPokemonFilter (pokemonId, show) {
-    pokemonFilter[pokemonId].show = show;
+function addPokemonFilter (pokemonIdString, pokemonId, formId, show) {
+    pokemonFilter[pokemonIdString].show = show;
     store('pokemon_filter', JSON.stringify(pokemonFilter));
 
     $.each(pokemonMarkers, function (index, pokemon) {
-        if (pokemon.pokemon_id === pokemonId) {
+        if (pokemon.pokemon_id === pokemonId && pokemon.form_id === formId) {
```
Comment (versx) [inline src/views/index-js.mustache:2703]:
```suggestion
        if (pokemon.pokemon_id === pokemonId && pokemon.form === formId) {
```

**Comment by Mygod** [comment]
Huh I thought I saw `form_id` in the output but I will double check later.

**Comment by versx** [comment]
> Huh I thought I saw `form_id` in the output but I will double check later.
https://github.com/RealDeviceMap/RealDeviceMap/blob/master/Sources/RealDeviceMap/Modell/Pokemon.swift#L46

**Comment by Mygod** [comment]
Yeah sorry the change was untested and I am not very familiar with the codebase yet. 😅

**Comment by versx** [comment]
> Yeah sorry the change was untested and I am not very familiar with the codebase yet. 😅

No worries at all, I appreciate the PR regardless.

File: src/views/index-js.mustache (hunk 3 above)
Comment (Mygod) [reply src/views/index-js.mustache:2703]: Indeed. `form_id` was used for `quest_rewards`.

**Comment by Mygod** [comment]
Oh it worked?

**Comment by versx** [comment]
> Oh it worked?

It works for most lol, some Pokemon have form id's of `0` which I think might be an RDM issue, which doesn't add some Pokemon to the exclude filter.

**Pull request merged**